src/Ledger/Foreign/LedgerTypes.agda

```diff
@@ -47,53 +50,70 @@ data ComputationResult E A : Set where
 
 HSMap : Set → Set → Set
 HSMap K V = List (Pair K V)
+Rational = Pair ℤ ℕ
 
 Coin          = ℕ
 Addr          = ℕ -- just payment credential
 
 TxId          = ℕ
 Ix            = ℕ
 Epoch         = ℕ
+ScriptHash    = ℕ
 
 AuxiliaryData = ⊤
 DataHash      = ⊤
 Datum         = ⊤
 Redeemer      = ⊤
 Anchor        = ⊤
+Network       = ⊤
+PParamsUpdate = ⊤
+Script        = ⊥
 
 TxIn          = Pair TxId Ix
-TxOut         = Pair Addr $ Pair Coin $ Maybe DataHash
+TxOut         = Pair Addr $ Pair Coin $ Pair (Maybe (Either Datum DataHash)) $ Maybe Script
 UTxO          = HSMap TxIn TxOut
 
 Hash          = ℕ
 
-data Tag : Set where Spend Mint Cert Rewrd Vote Propose : Tag
+GovActionID   = Pair TxId ℕ
+
+HashProtected : Set → Set
+HashProtected A = Pair A GovActionID
+
+data Tag : Set where Spend Mint Cert Rewrd VoteTag Propose : Tag
 RdmrPtr = Pair Tag Ix
 ExUnits = Pair ℕ ℕ
+ProtVer = Pair ℕ ℕ
 
 {-# FOREIGN GHC
+  type Rational = (Integer, Integer)
   type Coin  = Integer
   type Addr  = Integer
 
   type TxId    = Integer
   type Ix      = Integer
   type Epoch   = Integer
+  type ScriptHash    = Integer
 
   type AuxiliaryData = ()
   type DataHash      = ()
   type Datum         = ()
   type Redeemer      = ()
   type Anchor        = ()
+  type Network       = ()
+  type Script        = AgdaEmpty
```

Comment:
are you able to get this to compile now?

Reply:
Yes! 🥳